### PR TITLE
qa/suites/upgrade/telemetry-upgrade: add new telemetry upgrade tests

### PR DESCRIPTION
**Pull request by ljflores**
Example command:
```
./teuthology/virtualenv/bin/teuthology-suite -v -m smithi -c main -s upgrade:telemetry-upgrade --suite-repo https://github.com/ljflores/ceph.git --suite-branch wip-upgrade-telemetry -p 70
```

Signed-off-by: Laura Flores <lflores@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [x] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by ljflores** [comment]
New tracker created; would ideally like to handle this case: https://tracker.ceph.com/issues/57709

**Comment by ljflores** [comment]
Current runs:
```
./teuthology/virtualenv/bin/teuthology-suite -v -m smithi -c main -S 652bf75409c45c34ae939b241b00adba20f12869 -s upgrade:telemetry-upgrade --suite-repo https://github.com/ljflores/ceph.git --suite-branch wip-upgrade-telemetry -p 70
```
http://pulpito.front.sepia.ceph.com/lflores-2022-09-29_05:03:04-upgrade:telemetry-upgrade-main-distro-default-smithi/

**Comment by ljflores** [comment]
http://pulpito.front.sepia.ceph.com/lflores-2022-10-06_17:39:13-upgrade:telemetry-upgrade-main-distro-default-smithi/

**Comment by ljflores** [comment]
Successful runs:
```
./teuthology/virtualenv/bin/teuthology-suite -v -m smithi -c main -s upgrade:telemetry-upgrade --suite-repo https://github.com/ljflores/ceph.git --suite-branch wip-upgrade-telemetry -p 70
```
http://pulpito.front.sepia.ceph.com/lflores-2022-10-08_00:19:29-upgrade:telemetry-upgrade-main-distro-default-smithi/

**Comment by ljflores** [comment]
Latest runs; all green:

http://pulpito.front.sepia.ceph.com/lflores-2022-10-10_15:40:27-upgrade-main-distro-default-smithi/

**Comment by ljflores** [comment]
@yaarith the above tests are the most recent. I added coverage in our own `telemetry-upgrade` section, as well as in the existing upgrade tests.

File: qa/workunits/test_telemetry_pacific.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash -ex
```
Comment (ljflores) [inline qa/workunits/test_telemetry_pacific.sh:1]: This is the script that runs on Pacific before the upgrade

File: qa/workunits/test_telemetry_pacific_x.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
+#!/bin/bash -ex
```
Comment (ljflores) [inline qa/workunits/test_telemetry_pacific_x.sh:1]: This is the script that runs after we upgrade from Pacific to Reef (dev)

File: qa/workunits/test_telemetry_quincy.sh [hunk 3]
```diff
@@ -0,0 +1,44 @@
+#!/bin/bash -ex
```
Comment (ljflores) [inline qa/workunits/test_telemetry_quincy.sh:1]: This is the script that runs on Quincy before the upgrade

File: qa/workunits/test_telemetry_quincy_x.sh [hunk 4]
```diff
@@ -0,0 +1,40 @@
+#!/bin/bash -ex
```
Comment (ljflores) [inline qa/workunits/test_telemetry_quincy_x.sh:1]: This is the script that runs after we upgrade from Quincy to Reef (dev)

**Review by neha-ojha — APPROVED** [review]
I am unable to see the logs for the passed test http://pulpito.front.sepia.ceph.com/lflores-2022-10-10_15:40:27-upgrade-main-distro-default-smithi/, but LGTM!

**Pull request merged**